source/usage-examples/watch.txt

```diff
@@ -0,0 +1,139 @@
+- :manual:`aggregation stages </changeStreams/#modify-change-stream-output>`
+- :java-sync-api:`ChangeStreamIterable <com/mongodb/client/ChangeStreamIterable.html>`
+- :java-sync-api:`MongoCollection.watch() <com/mongodb/client/MongoCollection.html#watch()>`
+- :java-sync-api:`MongoDatabase.watch() <com/mongodb/client/MongoDatabase.html#watch()>`
+- :java-sync-api:`MongoClient.watch() <com/mongodb/client/MongoClient.html#watch()>`
+
+Process the Change Stream Events
+--------------------------------
+
+To capture events from a change stream, call the ``forEach()`` method
+with a callback function as shown below:
+
+.. code-block:: java
+
+   changeStream.forEach(event -> System.out.println("Change observed: " + event));
+
+The callback function triggers when a change event is emitted. You can
+specify logic in the callback to process the event document when it is
+received.
+
+.. admonition:: forEach() blocks the current thread
+   :class: important
+
+   Calls to ``forEach()`` block the current thread as long as the
+   corresponding change stream listens for events. If your program
+   needs to continue executing other logic, such as processing requests or
+   responding to user input, consider creating and listening to your
+   change stream in a separate thread.
+
+.. note::
+
+   For update operation change events, change streams only return the modified
+   fields by default rather than the full updated document. You can configure
+   your change stream to also return the most current version of the document
+   by calling the ``fullDocument()`` member method of the ``ChangeStreamIterable``
+   object with the value ``FullDocument.UPDATE_LOOKUP`` as follows:
+
+   .. code-block:: java
+
+      ChangeStreamIterable<Document> changeStream = database.watch()
+                .fullDocument(FullDocument.UPDATE_LOOKUP);
+
+Example
+-------
+
+The following example uses two separate applications to demonstrate how
+to listen for changes using a change stream:
+
+- The first application, named ``Watch``, opens a change stream
+  on the ``movies`` collection in the ``sample_mflix`` database.
+  ``Watch`` uses a callback to receive and print change events
+  that occur on the collection.
+
+- The second application, named ``WatchCompanion``, inserts a
+  single document into the ``movies`` collection in the ``sample_mflix``
+  database.
+
+First, run ``Watch`` to open the change stream on the collection and
+define a callback on the change stream using the ``forEach()`` method.
+Next, run ``WatchCompanion`` to generate a change event by performing
+a change to the collection.
+
+.. include:: /includes/connect-guide-note.rst
+
+``Watch.java``:
+
+.. literalinclude:: /includes/usage-examples/code-snippets/Watch.java
+   :language: java
+
+``WatchCompanion.java``:
+
+.. literalinclude:: /includes/usage-examples/code-snippets/WatchCompanion.java
+   :language: java
+
+If you run the example above, you should see output similar to the
```

Comment:
Nitpick: "example" here is slightly ambiguous. Could we say something like this:
"If you run each of the applications in sequence, you should see output from the Watch application that is similar to the following:"
The WatchCompanion app also prints out some information, so you might want to include that here as well.